src/web/extractors.rs

```diff
@@ -500,4 +630,65 @@ mod tests {
             .unwrap();
         assert_eq!(result, PreConditionHeader::IfUnmodifiedSince(32.14));
     }
+
+    #[test]
+    fn valid_header_with_valid_path() {
+        let payload = HawkPayload {
+            expires: Utc::now().timestamp() as f64 + 200000.0,
+            node: "friendly-node".to_string(),
+            salt: "saltysalt".to_string(),
+            user_id: 1,
+        };
+        let state = make_state();
+        let header = create_valid_hawk_header(
+            &payload,
+            &state,
+            "GET",
+            "/storage/1.5/1/storage/col2",
+            "localhost",
+            5000,
+        );
+        let req = TestRequest::with_state(state)
+            .header("authorization", header)
+            .method(Method::GET)
+            .uri("http://localhost:5000/storage/1.5/1/storage/col2")
+            .param("uid", "1")
+            .finish();
+        let result = HawkIdentifier::extract(&req).unwrap();
+        assert_eq!(result.legacy_id, 1);
+    }
+
+    #[test]
+    fn valid_header_with_invalid_uid_in_path() {
+        let payload = HawkPayload {
+            expires: Utc::now().timestamp() as f64 + 200000.0,
+            node: "friendly-node".to_string(),
+            salt: "saltysalt".to_string(),
+            user_id: 1,
+        };
+        let state = make_state();
+        let header = create_valid_hawk_header(
+            &payload,
+            &state,
+            "GET",
+            "/storage/1.5/5/storage/col2",
+            "localhost",
+            5000,
+        );
+        let req = TestRequest::with_state(state)
+            .header("authorization", header)
+            .method(Method::GET)
+            .uri("http://localhost:5000/storage/1.5/5/storage/col2")
+            .param("uid", "5")
```

Comment:
Do you have to repeat the `5` here because `TestRequest` won't parse the uid out of the supplised URI?

Reply:
Yup, TestRequest assumes the params were parsed by the Server and put in the the params object on the request.